01Matrix.js

```diff
@@ -0,0 +1,143 @@
+/*
+So we're going to represent our input as a grid where we're only concerned about the 1's, since we're trying 
+to find the distance for each 1 that is the closest to the nearest 0. 
+
+For this approach, we're going to use a multi-sourced BFS to find the distance to the nearest 0 
+and the way that works is we're going to gather the location of all the 0's first and add them to the queue,
+that way we can process them together. During the iteration, we're going to BFS each of those 
+locations in all 4 directions and until we reach a 1. Once we reach a 1, we mark it with the current 
+traveled distance so far.
```

Comment:
Why is it okay to do this? Can we reach the same 1 multiple times? How do we handle that? How are you making sure that it is okay to mark the cell with the current traveled difference?

Reply:
It's okay to do this because there are multiple 0's that we're focused on and by using a multi-sourced BFS, we can keep track of all the visited locations. Since a BFS travels by levels, we can mark the 1's as visited and with the closest path to a 0.

We can reach a 1 multiple times but with a visited set, we're avoiding locations that have been previously visited. Every time we visit a 1 for the first time,  it will be the shortest path from the 0 and every time a BFS travels by each level, we're adding 1 to current distance traveled so far.

Reply:
Good.